source/grid/tria.cc

```diff
@@ -8996,28 +8996,80 @@ Triangulation (const Triangulation<dim, spacedim> &other)
 
 
 
+#ifdef DEAL_II_WITH_CXX11
+template <int dim, int spacedim>
+Triangulation<dim, spacedim>::
+Triangulation (Triangulation<dim, spacedim> &&tria)
+  :
+  Subscriptor(tria),
+  smooth_grid(tria.smooth_grid),
+  periodic_face_pairs_level_0(std::move(tria.periodic_face_pairs_level_0)),
+  periodic_face_map(std::move(tria.periodic_face_map)),
+  levels(std::move(tria.levels)),
+  faces(std::move(tria.faces)),
+  vertices(std::move(tria.vertices)),
+  vertices_used(std::move(tria.vertices_used)),
+  manifold(std::move(tria.manifold)),
+  anisotropic_refinement(tria.anisotropic_refinement),
+  check_for_distorted_cells(tria.check_for_distorted_cells),
+  number_cache(tria.number_cache),
+  vertex_to_boundary_id_map_1d(tria.vertex_to_boundary_id_map_1d),
+  vertex_to_manifold_id_map_1d(tria.vertex_to_manifold_id_map_1d)
+{
+  for (unsigned int i=0; i<tria.levels.size(); ++i)
+    tria.levels[i] = nullptr;
+
+  tria.faces = nullptr;
+
+  tria.number_cache = internal::Triangulation::NumberCache<dim>();
+
+  tria.vertex_to_boundary_id_map_1d = nullptr;
+  tria.vertex_to_manifold_id_map_1d = nullptr;
+}
```

Comment:
Probably worth pulling the bulk of the invalidation of the input into a function so it can be reused when a move assignment operator is added.

Adding a move constructor to `NumberCache<dim>` would also make this a little nicer.